src/Filters/atoms/DateTimeBlock.tsx

```diff
@@ -125,16 +125,27 @@ const DateTimeBlock : React.FC<IProps> = ({
   const [displayMinutes, setDisplayMinutes] = useState<string>(getFormattedTime(convertMinutes));
 
   const setDateHours = useCallback(({target: {value}}: React.ChangeEvent<HTMLInputElement>) => {
-    const hourRegex  = /^[0-1]{0,1}[0-9]{0,1}$|(^2[0-4])$/;
-    if (!hourRegex.test(value)) {
-      return;
+    let newVal = '0';
+    if(value.length === 1){
+      if(value === '0'){
+        newVal = ('0');
+      } else {
+        newVal = ('0' + value).slice(-2) ;
+      }
+    } else {
+      newVal = value.slice(-2);
+    }
+    if(Number(newVal) > 24){
```

Comment:
After adding single digit if I enter next which makes whole number greater than 24 then 24 will be explicitly added, in this scenario we should not allow user to enter, for ex- if I enter 04 then if I press 0, then it will become 24, instead 0 shouldn't be allowed there.

Reply:
Done